packages/toolkit/src/lib/react-query-service/metric/usePipelineTriggerRecords.ts

```diff
@@ -19,8 +19,18 @@ export function usePipelineTriggerRecords({
   retry?: false | number;
   filterId: Nullable<string>;
 }) {
+  const queryKey = ["metrics", "pipelines", "triggers"];
+
+  if (filter) {
+    queryKey.push(filter);
+  }
```

Comment:
Since this filter has timeStamp, it will constantly changing. 

- First render: 2024T....-00:09:00 -> Request -> Trigger second render
- Second render: 2024T....-00:09:02

It will form a endless render, this is the reason I remove it in previous fix and use filterId

Reply:
Makes sense. Fixed my code.